src/s3fs.cpp

```diff
@@ -492,11 +456,7 @@ static int get_object_attribute(const char* path, struct stat* pstbuf, headers_t
     // (ex. getxattr() is called while writing to the opened file.)
     //
     if(add_no_truncate_cache || 0 != StatCache::getStatCacheData()->GetCacheSize()){
-        // add into stat cache
-        if(!StatCache::getStatCacheData()->AddStat(strpath, (*pheader), forcedir, add_no_truncate_cache)){
-            S3FS_PRN_ERR("failed adding stat cache [path=%s]", strpath.c_str());
-            return -ENOENT;
-        }
+        // readdir_multi_head calls AddStat but check it again
```

Comment:
The original code called AddStat again because it took into account when `add_no_truncate_cache` was `true`.
There is a problem with the original logic, but in `readdir_multi_head`(`multi_head_callback`) the AddStat call is always `no_truncate=false`.

So, if `add_no_truncate_cache=true`, I think you need to call AddStat again or review the logic.